apm-lambda-extension/docs/aws-lambda-extension.asciidoc

```diff
@@ -0,0 +1,137 @@
+[[aws-lambda-extension]]
+= AWS Lambda Extension (Experimental)
+
+Elastic's APM Language Agents are able to instrument AWS Lambda functions via an AWS Lambda Extension.
+
+[[aws-lambda-arch]]
+== Extension Architecture
+
+Normally, during the execution of a Lambda function, there's only a single language process running in the AWS Lambda execution environment.  However, with an AWS Lambda Extension, Lambda users can run a _second_ process along side their main service/application process.
+
+// TODO: replace below with a diagram diagram
+
+[source,txt]
+----
+    [Lamnda Function/Service] -> [Lambda Extension] -> [APM Server]
+----
+
+By using a custom built AWS Lambda Extension, Elastic APM Agents can send data to a locally running Lambda Extension process, and that process will forward data on to APM Server.  The Lambda Extension ensures that any latency between the Lambda Function and the AWS Server instance will not cause latency in the Lambda Function/Service itself.
+
+[[aws-lambda-install]]
+== Installing the Lambda Extension
```

Comment:
We don't have to add it now, but I'd like to eventually see an overview of all of the steps required at the start of this guide.